control-plane/agents/src/bin/ha/node/server.rs

```diff
@@ -65,24 +65,36 @@ impl NodeAgentSvc {
 }
 
 /// Disconnect cached NVMe controller.
-fn disconnect_controller(ctrlr: &NvmeController) -> Result<(), SvcError> {
+fn disconnect_controller(ctrlr: &NvmeController, new_path: String) -> Result<(), SvcError> {
+    let parsed_path = parse_uri(new_path.as_str())?;
+
     match get_nvme_path_buf(&ctrlr.path) {
         Some(pbuf) => {
             let subsystem = Subsystem::new(pbuf.as_path()).map_err(|_| SvcError::Internal {
                 details: "Failed to get NVMe subsystem for controller".to_string(),
             })?;
 
-            tracing::info!(
+            if subsystem.address
+                == SubsystemAddr::new(parsed_path.host(), parsed_path.port()).string_value()
```

Comment:
You can impl Eq for SubsystemAddr so you don't need to call string_value explicitly

Reply:
Right. Would do that. Thank you